### PR TITLE
[Jetpack Compose] Migrate offline area list to Compose Lazy Column

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1795

- Replaces RecyclerView with LazyColumn
- Renames OfflineAreaListItemViewModel to OfflineAreaDetails

@anandwana001  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.google.android.ground.R
+import com.google.android.ground.ui.theme.AppTheme
+
+@Composable
+fun OfflineAreaListItem(
+  modifier: Modifier,
```
Comment (anandwana001) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt:46]:
```suggestion
  modifier: Modifier = Modifier,
```

We can do this for less work for calling site

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          .padding(start = 16.dp, top = 4.dp, end = 24.dp, bottom = 4.dp)
+          .clickable { itemClicked(offlineAreaDetails.id) },
+      horizontalArrangement = Arrangement.spacedBy(16.dp, Alignment.CenterHorizontally),
+      verticalAlignment = Alignment.CenterVertically,
+    ) {
+      Icon(
+        imageVector = ImageVector.vectorResource(id = R.drawable.ic_offline_pin),
+        contentDescription = stringResource(id = R.string.offline_area_list_item_icon),
+        tint = MaterialTheme.colorScheme.primary,
+        modifier = Modifier.width(24.dp).height(24.dp),
```
Comment (anandwana001) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt:64]:
```suggestion
        modifier = Modifier.size(24.dp),
```

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt [hunk 3]
```diff
@@ -51,13 +59,24 @@ class OfflineAreasFragment : AbstractFragment() {
...
 
-    val recyclerView = binding.offlineAreasList
-    recyclerView.setHasFixedSize(true)
-    recyclerView.layoutManager = LinearLayoutManager(context)
-    recyclerView.adapter = offlineAreaListAdapter
     return binding.root
   }
+
+  @Composable
+  private fun ShowOfflineAreas() {
```
Comment (anandwana001) [inline ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt:70]: I am thinking to keep all composable functions outside fragments, we can have a separate file.

File: ground/src/test/java/com/google/android/ground/util/recyclerview/RecyclerViewMatcher.kt [hunk 4]
```diff
@@ -1,57 +0,0 @@
...
-package com.google.android.ground.util.recyclerview
-
-import android.content.res.Resources
-import android.view.View
-import androidx.recyclerview.widget.RecyclerView
-import org.hamcrest.Description
-import org.hamcrest.Matcher
-import org.hamcrest.TypeSafeMatcher
-
-fun atPositionOnView(recyclerViewId: Int, position: Int, targetViewId: Int): Matcher<View> =
```
Comment (anandwana001) [inline ground/src/test/java/com/google/android/ground/util/recyclerview/RecyclerViewMatcher.kt:25]: but we still have recyclerView in codebase? 

Or should we start migrating all to LazyColumn?

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt (hunk 3 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreasFragment.kt:70]: We already have a separate file which contains the main logic for how the item looks.

File: ground/src/test/java/com/google/android/ground/util/recyclerview/RecyclerViewMatcher.kt (hunk 4 above)
Comment (shobhitagarwal1612) [reply ground/src/test/java/com/google/android/ground/util/recyclerview/RecyclerViewMatcher.kt:25]: As per https://github.com/google/ground-android/issues/1795, we should migrate all layout based code to jetpack compose.

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/offlineareas/OfflineAreaListItem.kt]: Done

**Review by anandwana001 — APPROVED** [review]

**Pull request merged**